src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -231,6 +231,7 @@ object Group {
       group is noAppsAndGroupsWithSameName
       group is conditional[Group](_.id.isRoot)(noCyclicDependencies)
       group.groups is every(valid(validNestedGroup(group.id.canonicalPath(base))))
+      group is servicePortsDontOverlap
```

Comment:
AFAIK @unterstein moved the validation to deployment plan, in order to allow users to make changes to the root group without having to fix all existing applications, see #4167